### ISSUE TITLE
Assets within directories are not processed in a cross-platform way (problems with slashes)

### ISSUE THREAD

**Issue by yeputons** [sdarnell/meteor-win#27]
Demonstration: https://github.com/yeputons/meteor-windows-assets . I've tested it on Meteor 0.7.0.1, installed with `LaunchMeteor.exe`, no MSI.

Brief description:

```
Assets.getText('directory\\file.txt'); // Works OK
Assets.getText('directory/file.txt'); // Fails with 'Unknown asset' error
```

Problem is `/tools/server/boot.js:142`: `fileInfo.assets`'s keys are stored with back slashes (Windows' native method, all WinAPI functions use them), but `getText`'s argument has forward slashes (as it's on most UNIX systems). That produces compatibility problem: JS has no idea that `directory/file.txt` and `directory\file.txt` are the same.

Probably, one of these two should be changed to replace all forward slashes to back slashes (or vice-versa).

@sdarnell Please share your thoughs on it.

**Comment by yeputons** [comment]
For example, this affects https://github.com/arunoda/meteor-fast-render and https://github.com/oortcloud/atmosphere , which uses the former.

**Comment by sdarnell** [comment]
The issue is actually internal to meteor as node generally accept either form of slash (indeed some of the Win32 APIs do too).

I suspect that I have missed one of the places that process assets, or a new use has been added in 0.7.x.
Thanks for the initial diagnosis, I'll have to check but I suspect it may be easier to change assets to use forward slashes.

I'll have a look - it shouldn't be difficult to fix.

**Comment by sdarnell** [comment]
Fix submitted 47dd2a1af8949538c5c35204178899d474fdf264 and will be released shortly.
The issue was actually in the bundling of assets.

But note that the Assets.getText/getBinary() should always be passed paths with forward slashes, so after this change, paths with backslashes will no longer work.

**Comment by yeputons** [comment]
@sdarnell cool, thank you.
